contracts/v3/converters/General2Converter.sol

```diff
@@ -0,0 +1,209 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.6.12;
+
+import '@openzeppelin/contracts/token/ERC20/IERC20.sol';
+import '@openzeppelin/contracts/math/SafeMath.sol';
+import '@openzeppelin/contracts/token/ERC20/ERC20.sol';
+import '@openzeppelin/contracts/token/ERC20/SafeERC20.sol';
+
+import '../interfaces/IConverter.sol';
+import '../interfaces/IManager.sol';
+import '../interfaces/ICurve2Pool.sol';
+
+/**
+ * @title General2Converter
+ */
+contract General2Converter is IConverter {
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+
+    IManager public immutable override manager;
+    ICurve2Pool public immutable stableSwap2Pool;
+    IERC20 public immutable tokenCRV; // 2Crv
+
+    IERC20[2] public tokens;
+
+    mapping(address => int128) internal indices;
+
+    /**
+     * @param _tokens A list of the address of the token in the Curve pool
+     * @param _tokenCRV The address of the CRV token
+     * @param _stableSwap2Pool The address of 2Pool
+     * @param _manager The address of the Vault Manager
+     */
+    constructor(
+        IERC20[] memory _tokens,
+        IERC20 _tokenCRV,
+        ICurve2Pool _stableSwap2Pool,
+        IManager _manager
+    ) public {
+        require(_tokens.length == 2, 'Token count must be 2');
```

Comment:
The 2 versus 3 is the only difference between
General2Converter.sol and General3Converter.sol
So you might as well supply that as a parameter in the constructor and only have 1 code base to maintain

Reply:
The swap pool contract is different, so we need to have have 2 contracts

Reply:
You could have 1 source which you deploy twice, which could also be confusing. You could also put most of the code in a library and have 2 contracts including the library.
I've frequently seen bugs where code is copy pasted and something is changed in one copy but not the other. So having just 1 code base reduces risk.

Reply:
might be good to make sure the _coinCount is either 2 or 3